internal/services/monitor/monitor_diagnostic_setting_resource.go

```diff
@@ -462,6 +463,12 @@ func resourceMonitorDiagnosticSettingRead(d *pluginsdk.ResourceData, meta interf
 		return err
 	}
 
+	if strings.Contains(strings.ToLower(id.ResourceUri), "microsoft.kusto/clusters/") {
+		if v, err := kustoParse.ClusterIDInsensitively(id.ResourceUri); err == nil {
+			id.ResourceUri = v.ID()
+		}
+	}
```

Comment:
This can probably be simplified to 
```suggestion
	if v, err := kustoParse.ClusterIDInsensitively(id.ResourceUri); err == nil {
		id.ResourceUri = v.ID()
	}
```

Reply:
done.